hudi-client/hudi-client-common/src/main/java/org/apache/hudi/io/HoodieMergeHandleWithChangeLog.java

```diff
@@ -103,7 +103,7 @@ protected void writeInsertRecord(HoodieRecord<T> newRecord) throws IOException {
     // TODO Remove these unnecessary newInstance invocations
     HoodieRecord<T> savedRecord = newRecord.newInstance();
     super.writeInsertRecord(newRecord);
-    if (!HoodieOperation.isDelete(newRecord.getOperation())) {
+    if (!HoodieOperation.isDelete(newRecord.getOperation()) && !savedRecord.isDelete(schema, config.getPayloadConfig().getProps())) {
```

Comment:
Makes sense.  I'm wondering if other places using the same statement (`HoodieOperation.isDelete(newRecord.getOperation())`) should be updated too, e.g., in `writeUpdateRecord`.

Reply:
Potential fixes to other places should be in a separate PR.

Reply:
Should we write the delete into the change log, I think we should do that.

Reply:
I think we should (i.e., adding `else` block to handle the deletes).  However, it's a different issue we need to tackle.  I'll follow up.